src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,262 @@
+package duke;
+
+import duke.command.Deadline;
+import duke.command.Event;
+import duke.command.Todo;
+import duke.task.Task;
+
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.IOException;
+import java.io.FileWriter;
+import java.nio.file.Paths;
+
+public class Duke {
+
+    public static int taskNum = 0;
+    public static ArrayList<Task> tasks = new ArrayList<>();
+
+    public static final String root = System.getProperty("user.dir");
+    public static final Path dataPath = Paths.get(root, "data", "duke.txt");
+    public static final Path dirPath = Paths.get(root, "data");
+
+    public static void main(String[] args) {
+        String userCommand;
+        Scanner userType = new Scanner(System.in);
+        greet();
+        try {
+            loadData(dataPath);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        do {
+            // Read in the keyboard input from user, and refer to different conditions
+            userCommand = userType.nextLine();
+            if (userCommand.equals("bye")) {
+                break;
+            } else if (userCommand.equals("list")) {
+                printList();
+            } else if (userCommand.contains("done")) {
+                int id = Integer.parseInt(userCommand.substring(5));
+                taskDone(tasks.get(id - 1));
+            } else if (userCommand.startsWith("todo")) {
+                addTodo(userCommand);
+            } else if (userCommand.startsWith("event")) {
+                addEvent(userCommand);
+            } else if (userCommand.startsWith("deadline")) {
+                addDeadline(userCommand);
+            } else if (userCommand.startsWith("delete")) {
+                deleteTask(userCommand);
+            } else {
+                printSign();
+                System.out.println(" ☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+                printSign();
+            }
+        } while (!Objects.equals(userCommand, "bye"));
+        exit();
+        saveData();
+    }
+
+    public static void loadData(Path dataPath) throws IOException {
+        File dataDirectory = new File(dirPath.toString());
```

Comment:
This method is too long ([>30 LoC](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods)). You could review and see if there are blocks of code that could be abstracted out to methods.